tests/integration/test_dbapi_integration.py

```diff
@@ -1120,3 +1121,27 @@ def test_prepared_statements(run_trino):
     cur.execute('DEALLOCATE PREPARE test_prepared_statements')
     cur.fetchall()
     assert cur._request._client_session.prepared_statements == {}
+
+
+def test_set_timezone_in_connection(run_trino):
+    _, host, port = run_trino
+
+    trino_connection = trino.dbapi.Connection(
+        host=host, port=port, user="test", catalog="tpch", timezone="Europe/Brussels"
+    )
+    cur = trino_connection.cursor()
+    cur.execute('SHOW TABLES FROM information_schema')
+    cur.fetchall()
+    assert cur._request.http_headers[constants.HEADER_TIMEZONE] == "Europe/Brussels"
```

Comment:
instead of the request header testing, please check using `SELECT current_timezone()`